src/dune/file_tree.ml

```diff
@@ -521,10 +521,8 @@ let files_of path =
   match find_dir path with
   | None -> Path.Source.Set.empty
   | Some dir ->
-    Path.Source.Set.of_list
-      (List.map
-         (String.Set.to_list (Dir0.files dir))
-         ~f:(Path.Source.relative path))
+    Dir0.files dir |> String.Set.to_list
+    |> Path.Source.Set.of_list_map ~f:(Path.Source.relative path)
```

Comment:
It might be worth to introduce a `Path.Source.Set.relative : Path.Source.t -> String.Set.t -> Path.Source.Set.t` function soon. it could already be implemented in terms of `String.Set.map`, and if we switch to a trie representation, it would become really cheap.